src/core/document.js

```diff
@@ -482,7 +479,9 @@ class Page {
     return this._parsedAnnotations.then(function (annotations) {
       const annotationsData = [];
       for (let i = 0, ii = annotations.length; i < ii; i++) {
-        if (!intent || isAnnotationRenderable(annotations[i], intent)) {
+        // Get the annotation even if it's hidden because
+        // JS can change its display.
+        if (annotations[i].viewable) {
```

Comment:
Won't this be an observable API change as-is, since you're now return all viewable annotations *regardless* of the `intent`-value?
I believe that what you actually want here is something like this:
```javascript
if (!intent || (intent === "display" && annotations[i].viewable)) {
```